### ISSUE TITLE
& insertion errors (Qt5?)

### ISSUE THREAD

**Issue by bhallen** [PhonologicalCorpusTools/CorpusTools#479]
Michael M.:

I recently reinstalled my OS (now Kubuntu 15.04) and now PCT's GUI exhibits a strange behavior. For example (when trying to download the example corpus so I can address #470 ):

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/iogui.py", line 315, in accept
    name = self.corporaWidget.value()
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/widgets.py", line 2279, in value
    return self.options[w.text()]
KeyError: '&Example toy corpus'

...and (when trying to calculate functional load, after trying to add the segment pair 'm' and 'n' but ending up in the display window with '&m' and '&n')...

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/windows.py", line 234, in newTable
    self.calc()
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/windows.py", line 220, in calc
    kwargs = self.generateKwargs()
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/flgui.py", line 256, in generateKwargs
    alg = self.algorithmWidget.value()
  File "/usr/local/lib/python3.4/dist-packages/corpustools-1.1.0-py3.4.egg/corpustools/gui/widgets.py", line 2279, in value
    return self.options[w.text()]
KeyError: '&Minimal pairs'

So basically it looks like ampersands are getting prefixed to things all over the place. From what I can tell online, this seems to be an incompatibility of KDE/Kubuntu with Qt, fortunately, but I wanted to see whether this happens to look like a familiar problem to you or whether you might have any ideas for me. In the worst case scenario, I'll do my testing of the PCT GUI in a virtualenv.

**Comment by mmcauliffe** [comment]
Interesting!  Ampersands are used in Qt for shortcutting, so if you had a menu item "File" and you want it available via ctrl+F (or cmd+F on mac) you'd given the label "&File", and Qt would take care of the rest.  If Qt on KDE is appending those by default, then it runs into a slight issue where I've used the labels of Qt elements as keys in a dictionary.  Probably not the best practice on my part.

One way to fix it quickly, would be to track down and wrap any `w.text()` call in a sanitize method that does something like `def qt_sanitize(string): return string.replace('&', '')`.  I can't replicate it on Ubuntu 15.04, so it's seems like it's just KDE based distros.  Very odd!

**Comment by mmcauliffe** [comment]
Oh, I guess I use miniconda environments for everything, so it may be bugged in the system version of Qt5 on Ubuntu too...

**Comment by bhallen** [comment]
Thanks, Michae.! I sanitized the w.text() method calls and now I can at least download corpora successfully. This problem is rampant, though, so at some point in the future, I'll go through the dozens of .text() calls and sanitize them all. So I'll mark this as a low-priority bug for me to fix later.

**Comment by bhallen** [comment]
The newest version of develop, which will overwrite master, lacks my ad hoc fixes. Since I'm not even using Kubuntu anymore, I'm marking this issue closed as a will-not-fix.
